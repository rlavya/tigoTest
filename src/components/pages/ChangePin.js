import React, { Component } from "react";
import RaisedButton from "material-ui/RaisedButton";
import TextField from "material-ui/TextField";
import PageTitle from "../components/PageTitle";
import { lightBlue200, indigo300 } from "material-ui/styles/colors";
import get from "lodash/get";
import { connect } from "react-redux";
import {
  updateNewPinAction,
  updateOldPinAction,
  changePinAction
} from "../../actionCreators/auth.js";
import { onEnterAction } from "../../util";
import styled from "styled-components";

const InputWrap = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  img {
    display: block;
    margin-bottom: -23px;
    margin-right: 10px;
    margin-left: -30px;
  }
`;

export class ChangePin extends Component {
  render() {
    const changePinIfValid = onEnterAction(() => {
      if (this.props.oldPin.trim() !== "" && this.props.newPin.trim() !== "") {
        this.props.changePin();
      }
    });

    return (
      <PageTitle page="ChangePin">
        <div className="max-width-4 mx-auto">
          <div className="clearfix center">
            <h1 className="light text-primary h2">Ingresa tus credenciales:</h1>

            <div className="masked-input">
              <InputWrap>
                <img src="/images/icn-password.png" alt="" />
                <TextField
                  underlineFocusStyle={{ borderColor: lightBlue200 }}
                  floatingLabelStyle={{ color: indigo300 }}
                  floatingLabelText="PIN ACTUAL"
                  type="tel"
                  pattern="[0-9]*"
                  id="oldPin"
                  value={this.props.oldPin}
                  onChange={this.props.updateOldPin}
                  onKeyDown={changePinIfValid}
                />
              </InputWrap>
            </div>
            <br />
            <div className="masked-input">
              <InputWrap>
                <img src="/images/icn-password-new.png" alt="" />
                <TextField
                  underlineFocusStyle={{ borderColor: lightBlue200 }}
                  floatingLabelStyle={{ color: indigo300 }}
                  floatingLabelText="PIN NUEVO"
                  type="tel"
                  pattern="[0-9]*"
                  id="newPin"
                  value={this.props.newPin}
                  onChange={this.props.updateNewPin}
                  onKeyDown={changePinIfValid}
                />
              </InputWrap>
            </div>
            <br />

            <RaisedButton
              className="mt2"
              secondary={true}
              label="Cambiar"
              onClick={this.props.changePin}
              disabledBackgroundColor="#CAF1F6"
              disabledLabelColor="#FFF"
              disabled={
                this.props.oldPin.trim() === "" ||
                this.props.newPin.trim() === ""
              }
            />
          </div>
        </div>
      </PageTitle>
    );
  }
}

export default connect(
  state => {
    return {
      oldPin: get(state, "auth.pinChange.old") || "",
      newPin: get(state, "auth.pinChange.new") || ""
    };
  },
  dispatch => {
    return {
      updateOldPin: (e, value) => {
        return dispatch(updateOldPinAction(value));
      },
      updateNewPin: (e, value) => {
        return dispatch(updateNewPinAction(value));
      },
      changePin: (e, value) => {
        return dispatch(changePinAction());
      }
    };
  }
)(ChangePin);
