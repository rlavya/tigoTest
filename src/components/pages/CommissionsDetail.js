import { connect } from "react-redux";
import { siteTree } from "../../siteTree";
import { dataForId, configToId } from "../../util/db";
import { redirectTo } from "../../router";
import TransactionsOrCommissionsDetail from "../components/TransactionsOrCommissionsDetail";

export default connect(
  state => {
    const items = dataForId(
      state,
      configToId("CommissionDetail", "list", true)
    );
    return {
      items,
      pageName: siteTree.TransactionsDetail.name,
      periodFilter: state.route.periodFilter || "hora",
      transactionFilter: state.route.transactionFilter || "todas"
    };
  },
  dispatch => {
    return {
      changePeriodFilter(filter) {
        redirectTo({ periodFilter: filter }, true);
      },
      changeTransactionFilter(filter) {
        redirectTo({ transactionFilter: filter }, true);
      }
    };
  }
)(TransactionsOrCommissionsDetail);
