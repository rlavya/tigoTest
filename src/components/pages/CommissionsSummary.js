import { connect } from "react-redux";
import { siteTree } from "../../siteTree";
import { dataForId, configToId } from "../../util/db";
import TransactionsOrCommissionsSummary from "../components/TransactionsOrCommissionsSummary";

export default connect(state => {
  const commissions = dataForId(state, configToId("Commission", "list", true));

  return {
    items: commissions,
    pageName: siteTree.CommissionsSummary.name,
    detailViewPageName: siteTree.CommissionsDetail.name
  };
})(TransactionsOrCommissionsSummary);
