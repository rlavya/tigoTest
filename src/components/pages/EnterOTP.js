import React, { Component } from "react";
import PageTitle from "../components/PageTitle";
import { lightBlue200 } from "material-ui/styles/colors";
import RaisedButton from "material-ui/RaisedButton";
import get from "lodash/get";
import { connect } from "react-redux";
import TextField from "material-ui/TextField";
import styled from "styled-components";

import {
  requestOtpAction,
  updateOTPCodeAction,
  validateOTPCodeAction
} from "../../actionCreators/auth.js";

import { onEnterAction } from "../../util";

const TextFieldWrap = styled.div`
  margin: 0 auto;
  width: 157px;
`;

export class EnterOTP extends Component {
  render() {
    const onEnter = onEnterAction(() => {
      this.props.validateCode();
    });
    return (
      <PageTitle page="OTP">
        <div className="max-width-4 mx-auto">
          <div className="center">
            <h1 className="light text-primary h2">
              Ingresa el código <br />de verificación
            </h1>
            <div className="mt3">
              <p className="regular text-primary inline">
                Te enviamos un SMS al{" "}
              </p>
              <p className="bold text-primary inline">+{this.props.msisdn}</p>
            </div>
            <TextFieldWrap>
              <TextField
                value={this.props.otpCode}
                onChange={this.props.updateCode}
                id="validate-otp"
                className="mt4 mb1"
                type="tel"
                pattern="[0-9]*"
                fullWidth={true}
                onKeyDown={onEnter}
                underlineStyle={{
                  backgroundImage: "url(/images/segmented_inactive.png)",
                  backgroundRepeat: "no-repeat",
                  backgroundPosition: "2px 0px",
                  height: "2px",
                  border: "none",
                  bottom: "1px"
                }}
                underlineFocusStyle={{
                  backgroundImage: "url(/images/segmented_active.png)",
                  backgroundRepeat: "no-repeat",
                  backgroundPosition: "2px 0px",
                  height: "2px",
                  border: "none",
                  bottom: "1px"
                }}
                inputStyle={{
                  fontSize: "2.5rem",
                  fontFamily: "'Roboto Mono', monospace",
                  letterSpacing: "2px"
                }}
              />
            </TextFieldWrap>
            <br />
            <RaisedButton
              label="Continuar"
              secondary={true}
              className="mt2 mb3"
              disabledBackgroundColor="#CAF1F6"
              disabledLabelColor="#FFF"
              disabled={this.props.otpCode.length < 6}
              onClick={this.props.validateCode}
            />
            <br />
            <p className="inline text-primary mt2">¿No recibiste el código?</p>
            <p
              className="inline pointer"
              style={{ color: lightBlue200 }}
              onClick={this.props.requestNewCode}
            >
              {" "}
              Enviar de nuevo.
            </p>
          </div>
        </div>
      </PageTitle>
    );
  }
}

export default connect(
  state => {
    return {
      msisdn: get(state, "auth.otp.msisdn") || "",
      otpCode: get(state, "auth.otpCode") || ""
    };
  },
  dispatch => {
    return {
      updateCode: (e, value) => {
        return dispatch(updateOTPCodeAction((value || "").trim()));
      },
      validateCode: () => {
        return dispatch(validateOTPCodeAction());
      },
      requestNewCode: () => {
        return dispatch(requestOtpAction({}));
      }
    };
  }
)(EnterOTP);
