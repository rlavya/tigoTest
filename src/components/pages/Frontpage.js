import React, { Component } from "react";
import { List, ListItem } from "material-ui/List";
import ChevronRight from "material-ui/svg-icons/navigation/chevron-right";
import Divider from "material-ui/Divider";
import { redirectTo } from "../../router";

export class Frontpage extends Component {
  render() {
    return (
      <div className="max-width-4 mx-auto bg-primary height100 flex flex-column">
        <img
          src="http://via.placeholder.com/250x250"
          width="250"
          height="250"
          className="mx-auto my2"
          alt=""
        />
        <List>
          <Divider style={{ backgroundColor: "#406A9C" }} />
          <ListItem
            primaryText={
              <div>
                <b>PDV</b> - Punto de Venta
              </div>
            }
            rightIcon={<ChevronRight color="#4CC5F2" />}
            onClick={() => {
              redirectTo({ page: "Login" });
            }}
            style={{ color: "#fafafa" }}
          />
          <Divider style={{ backgroundColor: "#406A9C" }} />
          <ListItem
            primaryText={
              <div>
                <b>PDA</b> - Punto de Activaci√≥n
              </div>
            }
            rightIcon={<ChevronRight color="#4CC5F2" />}
            onClick={() => {
              redirectTo({ page: "Login" });
            }}
            style={{ color: "#fafafa" }}
          />
          <Divider style={{ backgroundColor: "#406A9C" }} />
        </List>
      </div>
    );
  }
}

export default Frontpage;
