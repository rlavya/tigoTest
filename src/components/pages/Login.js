import React, { Component } from "react";
import RaisedButton from "material-ui/RaisedButton";
import TextField from "material-ui/TextField";
import PageTitle from "../components/PageTitle";
import { lightBlue200, indigo300 } from "material-ui/styles/colors";
import get from "lodash/get";
import { connect } from "react-redux";
import {
  updatePosPinAction,
  updatePosDMSIDAction,
  loginAction
} from "../../actionCreators/auth.js";
import styled from "styled-components";
import { urlFor } from "../../router";
import { onEnterAction } from "../../util";

const InputWrap = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  img {
    display: block;
    margin-bottom: -23px;
    margin-right: 5px;
    margin-left: -15px;
  }
`;

export class Login extends Component {
  render() {
    const loginIfValid = onEnterAction(() => {
      if (this.props.pin.trim() !== "" && this.props.dmsid.trim() !== "") {
        this.props.login();
      }
    });

    return (
      <PageTitle page="Login">
        <div className="max-width-4 mx-auto">
          <div className="clearfix center">
            <h1 className="light text-primary h2">Ingresa tus credenciales:</h1>
            <InputWrap>
              <img src="/images/icn-DMSID.png" alt="" />
              <TextField
                underlineFocusStyle={{ borderColor: lightBlue200 }}
                floatingLabelStyle={{ color: indigo300 }}
                floatingLabelText="DMS ID"
                id="dmsid"
                value={this.props.dmsid}
                onChange={this.props.updateDMSID}
                type="tel"
                pattern="[0-9]*"
                onKeyDown={loginIfValid}
                inputStyle={{ fontSize: "1.5rem" }}
              />
            </InputWrap>
            <InputWrap>
              <img src="/images/icn-password.png" alt="" />
              <div className="masked-input">
                <TextField
                  underlineFocusStyle={{ borderColor: lightBlue200 }}
                  floatingLabelStyle={{ color: indigo300 }}
                  floatingLabelText="PIN"
                  id="password"
                  type="tel"
                  pattern="[0-9]*"
                  value={this.props.pin}
                  onChange={this.props.updatePin}
                  onKeyDown={loginIfValid}
                  inputStyle={{ fontSize: "1.5rem" }}
                />
              </div>
            </InputWrap>
            <br />

            <RaisedButton
              className="mt2"
              secondary={true}
              label="Iniciar"
              onClick={this.props.login}
              disabledBackgroundColor="#CAF1F6"
              disabledLabelColor="#FFF"
              disabled={
                this.props.pin.trim() === "" || this.props.dmsid.trim() === ""
              }
            />
            <br />
            <p>
              <a
                href={urlFor({ page: "ChangePin" })}
                className="text-secondary text-decoration-none light"
              >
                Cambiar PIN
              </a>
            </p>
          </div>
        </div>
      </PageTitle>
    );
  }
}

export default connect(
  state => {
    return {
      dmsid: get(state, "auth.dmsid") || "",
      pin: get(state, "auth.pin") || ""
    };
  },
  dispatch => {
    return {
      updatePin: (e, value) => {
        return dispatch(updatePosPinAction(value));
      },
      updateDMSID: (e, value) => {
        return dispatch(updatePosDMSIDAction(value));
      },
      login: (e, value) => {
        return dispatch(loginAction());
      }
    };
  }
)(Login);
