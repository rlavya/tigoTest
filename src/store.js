import { applyMiddleware, createStore } from "redux";
import { createLogger } from "redux-logger";
import { composeWithDevTools } from "redux-devtools-extension/developmentOnly";
import thunk from "redux-thunk";
import reducer from "./reducer";

const getMiddleware = () => {
  return applyMiddleware(createLogger(), thunk);
};

const store = createStore(reducer, composeWithDevTools(getMiddleware()));

export default store;
