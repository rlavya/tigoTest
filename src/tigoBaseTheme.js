const tigoBaseTheme = {
  fontFamily: "Roboto, sans-serif",
  borderRadius: 2,
  palette: {
    primary1Color: "#00377B",
    primary2Color: "#4CC5F2",
    primary3Color: "#333",
    accent1Color: "#4CC5F2",
    accent2Color: "#4CC5F2",
    accent3Color: "#4CC5F2",
    textColor: "#00377B",
    secondaryTextColor: "#839ebf",
    alternateTextColor: "#fafafa",
    canvasColor: "#FFFFFF"
  }
};

export default tigoBaseTheme;
