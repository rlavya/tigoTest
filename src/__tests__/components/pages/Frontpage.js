import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import getMuiTheme from "material-ui/styles/getMuiTheme";
import tigoBaseTheme from "../../../tigoBaseTheme";
import { Frontpage } from "../../../components/pages/Frontpage";
import renderer from "react-test-renderer";
import React from "react";
import store from "../../../store";
import {Provider} from "react-redux";

it("renders correctly", () => {
  const rendered = renderer.create(
    <MuiThemeProvider muiTheme={getMuiTheme(tigoBaseTheme)}>
      <Provider store={store}>
        <Frontpage />
      </Provider>
    </MuiThemeProvider>
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});
