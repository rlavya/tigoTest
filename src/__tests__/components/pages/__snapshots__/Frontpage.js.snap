// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="max-width-4 mx-auto bg-primary height100 flex flex-column"
>
  <img
    alt=""
    className="mx-auto my2"
    height="250"
    src="http://via.placeholder.com/250x250"
    width="250"
  />
  <div
    style={
      Object {
        "muiPrepared": true,
        "padding": "8px 0px 8px 0px",
      }
    }
  >
    <hr
      style={
        Object {
          "backgroundColor": "#406A9C",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      }
    />
    <div>
      <span
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        open={null}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "backgroundColor": null,
            "border": 10,
            "boxSizing": "border-box",
            "color": "#fafafa",
            "cursor": "pointer",
            "display": "block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "fontWeight": "inherit",
            "lineHeight": "16px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "marginLeft": 0,
                "muiPrepared": true,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 56,
                "paddingTop": 16,
                "position": "relative",
              }
            }
          >
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#00377B",
                  "display": "block",
                  "fill": "#4CC5F2",
                  "height": 24,
                  "margin": 12,
                  "muiPrepared": true,
                  "position": "absolute",
                  "right": 4,
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
            <div
              style={
                Object {
                  "muiPrepared": true,
                }
              }
            >
              <b>
                PDV
              </b>
               - Punto de Venta
            </div>
          </div>
        </div>
      </span>
    </div>
    <hr
      style={
        Object {
          "backgroundColor": "#406A9C",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      }
    />
    <div>
      <span
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        open={null}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "backgroundColor": null,
            "border": 10,
            "boxSizing": "border-box",
            "color": "#fafafa",
            "cursor": "pointer",
            "display": "block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "fontWeight": "inherit",
            "lineHeight": "16px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "marginLeft": 0,
                "muiPrepared": true,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 56,
                "paddingTop": 16,
                "position": "relative",
              }
            }
          >
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#00377B",
                  "display": "block",
                  "fill": "#4CC5F2",
                  "height": 24,
                  "margin": 12,
                  "muiPrepared": true,
                  "position": "absolute",
                  "right": 4,
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
            <div
              style={
                Object {
                  "muiPrepared": true,
                }
              }
            >
              <b>
                PDA
              </b>
               - Punto de Activaci√≥n
            </div>
          </div>
        </div>
      </span>
    </div>
    <hr
      style={
        Object {
          "backgroundColor": "#406A9C",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      }
    />
  </div>
</div>
`;
